inference-engine/thirdparty/clDNN/src/network.cpp

```diff
@@ -470,25 +389,36 @@ void network_impl::allocate_primitives() {
     }
 }
 
-void network_impl::build_insts_deps() {
+void network::build_insts_deps() {
     for (auto& inst : _primitives) {
         inst.second->build_deps();
     }
 }
 
-void network_impl::build_exec_order() {
+void network::build_exec_order() {
     for (auto& node : _program->get_processing_order()) {
         if (!node->is_type<data>() && !(node->is_type<mutable_data>() && node->get_dependencies().empty())) {
             add_to_exec_order(node->id());
         }
     }
 }
-void network_impl::add_to_exec_order(const primitive_id& id) {
+void network::add_to_exec_order(const primitive_id& id) {
     auto inst = get_primitive(id);
     _exec_order.push_back(inst);
 }
 
-void network_impl::execute(const std::vector<event::ptr>& events) {
+std::map<primitive_id, network_output> network::execute(const std::vector<event::ptr>& dependencies) {
+    execute_impl(dependencies);
+
+    auto output_ids = get_output_ids();
+    std::map<primitive_id, network_output> result;
+    for (auto& id : output_ids) {
+        result.emplace(id, get_output(id));
+    }
+    return result;
+}
+
+void network::execute_impl(const std::vector<event::ptr>& events) {
```

Comment:
Wouldn't `execute_network` be a more appropriate name for the function?

Reply:
I think `network::execute_impl` is better than `network::execute_network`